wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/PluginFQNParser.java

```diff
@@ -57,60 +67,69 @@
                 + "Only one editor is supported per workspace.");
       }
-      metaFQNs.addAll(parsePluginFQNs(editor));
+      metaFQNs.addAll(editorsFQNs);
     }
     return metaFQNs;
   }
 
-  private Collection<PluginFQN> parsePluginFQNs(String... plugins) throws InfrastructureException {
+  private Collection<PluginFQN> parsePluginFQNs(String attribute) throws InfrastructureException {
+
+    String[] plugins = splitAttribute(attribute);
+    if (plugins.length == 0) {
+      return Collections.emptyList();
+    }
+
     List<PluginFQN> collectedFQNs = new ArrayList<>();
     for (String plugin : plugins) {
-      URI repo = null;
-      String idVersionString;
-      final int idVersionTagDelimiter = plugin.lastIndexOf("/");
-      idVersionString = plugin.substring(idVersionTagDelimiter + 1);
-      if (idVersionTagDelimiter > -1) {
-        try {
-          repo = new URI(plugin.substring(0, idVersionTagDelimiter));
-        } catch (URISyntaxException e) {
-          throw new InfrastructureException(
-              String.format(
-                  "Plugin registry URL is incorrect. Problematic plugin entry: %s", plugin));
-        }
-      }
-      String[] idAndVersion = idVersionString.split(":");
-      if (idAndVersion.length != 2 || idAndVersion[0].isEmpty() || idAndVersion[1].isEmpty()) {
+      PluginFQN pFQN = parsePluginFQN(plugin);
+
+      if (collectedFQNs.stream().anyMatch(p -> p.getId().equals(pFQN.getId()))) {
         throw new InfrastructureException(
-            String.format("Plugin format is illegal. Problematic plugin entry: %s", plugin));
+            format(
+                "Invalid Che tooling plugins configuration: plugin %s is duplicated",
+                pFQN.getId())); // even if different registries
       }
-      if (collectedFQNs
-          .stream()
-          .anyMatch(
-              p -> p.getId().equals(idAndVersion[0]) && p.getVersion().equals(idAndVersion[1]))) {
+      collectedFQNs.add(pFQN);
+    }
+    return collectedFQNs;
+  }
+
+  private PluginFQN parsePluginFQN(String plugin) throws InfrastructureException {
+    String registry;
+    String id;
+    URI registryURI = null;
+    Matcher matcher = PLUGIN_PATTERN.matcher(plugin);
+    if (matcher.matches()) {
+      registry = matcher.group("registry");
+      id = matcher.group("id");
+    } else {
+      throw new InfrastructureException(format(INCORRECT_PLUGIN_FORMAT_TEMPLATE, plugin));
+    }
+    if (!isNullOrEmpty(registry)) {
+      try {
+        registryURI = new URI(registry);
+      } catch (URISyntaxException e) {
         throw new InfrastructureException(
-            String.format(
-                "Invalid Che tooling plugins configuration: plugin %s:%s is duplicated",
-                idAndVersion[0], idAndVersion[1])); // even if different repos
+            format(
+                "Plugin registry URL '%s' is incorrect. Problematic plugin entry: '%s'",
```

Comment:
I'd use `invalid` instead of `incorrect` here, the latter suggests an issue contacting the registry.

Reply:
makes sense, changed